RoomDraw/app/helpers/users_helper.rb

```diff
@@ -1,2 +1,29 @@
 module UsersHelper
+  def student_room(sid)
+    Room.find_by_sql(query(sid)).first
+  end
+
+  def query(sid)
+    q = "SELECT rooms.*
+        FROM rooms, occupies
+        WHERE rooms.id = occupies.room_id AND occupies.student_id = #{sid}
+          AND academic_year = #{draw_year} AND "
+    q += sem_string
```

Comment:
Seems like this could be "cleaner" if we did something like 
```
sem = sem_string
q = "... = #{draw_year} AND #{sem}"
```

instead of the += business. That said, this works and is fewer LOCs.